packages/preview-middleware-client/src/cpe/changes/service.ts

```diff
@@ -96,6 +100,7 @@ export class ChangeService {
     private savedChanges: SavedPropertyChange[] = [];
     private sendAction: (action: ExternalAction) => void;
     private pendingChanges: PendingChange[] = [];
+    private changedFiles: ChangeFiles[] = [];
```

Comment:
This type is a bit misleading. We use it for change file objects which have a lot more properties, but we don't need to know the exact properties here. If we're not interested in the content and we just pass it through then it would better to use `unknown` type.

Also if we use record, then it would be easier to match it with the UI objects. 

```suggestion
    private changedFiles: Record<string, unknown> = {};
```

Reply:
I defined the type for `changedFiles` a little bit differently than you suggestion, as it was causing problems with another type.
Please check commit: 809dc7fe4bc59a58864a0d90cb067cf59eabd1c9

Reply:
After discussion, please check commit: 21ab914199b1a4a78bc4e6f532bcd631a369a916